vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -500,6 +501,11 @@ void ReactInstanceWin::Initialize() noexcept {
           std::unique_ptr<facebook::jsi::ScriptStore> scriptStore = nullptr;
           std::unique_ptr<facebook::jsi::PreparedScriptStore> preparedScriptStore = nullptr;
 
+          if (const auto jsExecutorFactory = Microsoft::JSI::JSExecutorFactorySettings::GetJSExecutorFactory(
+                  winrt::Microsoft::ReactNative::ReactPropertyBag(strongThis->Options().Properties))) {
+            devSettings->jsExecutorFactory = jsExecutorFactory;
+          }
+
           switch (m_options.JsiEngine()) {
```

Comment:
This probably needs to move back to an else, so we don't bother creating a RuntimeHolderLazyInit when it's not needed.